doc/modules/mod_vcard.md

```diff
@@ -3,47 +3,137 @@ This module provides support for vCards, as specified in [XEP-0054: vcard-temp](
 
-* `ldap_vcard_map` (list of `{VCardField, LDAPPattern, [LDAPField]}`, default: see description): Mappings between VCard and LDAP fields. For the default setting, please see `[MongooseIM root]/src/mod_vcard_ldap.erl`, line 74.
+###### `ldap_pool_tag`, `ldap_base`, `ldap_uids`, `ldap_filter`, `ldap_deref`
+* **Syntax:** see description
+* **Default:** see description
+* **Example:** see description
+
+These options are the same as for the [LDAP authentication module](../authentication-backends/LDAP-authentication-module.md#configuration-options).
 
-* `ldap_search_fields` (list of `{SearchField, LDAPField}`, default: see description): Mappings between the human-readable search fields and LDAP fields.
- For the default setting, please see `[MongooseIM root]/src/mod_vcard_ldap.erl`, line 96.
+###### `ldap_vcard_map`
+* **Syntax:** Array of TOML tables with the following keys: `"vcard_field"`, `"ldap_pattern"`, `"ldap_field"` and string values.
+* **Default:** see description
+* **Example:** `ldap_vcard_map = [{vcard_field = "FN", ldap_pattern = "%s", ldap_field = "displayName"}]`
 
-* `ldap_search_reported` (list of `{SearchField, VCardField}`, default: see description): Mappings between the human-readable search fields and VCard fields.
- For the default setting, please see `[MongooseIM root]/src/mod_vcard_ldap.erl`, line 109.
+Mappings between VCard and LDAP fields. For the default settings, please see `[MongooseIM root]/src/mod_vcard_ldap.erl`, line 79.
 
-* `ldap_search_operator` (`or` | `and`, default: `and`): A default operator used for search query items.
+###### `ldap_search_fields`
+* **Syntax:** Array of TOML tables with the following keys: `"search_field"`, `"ldap_field"` and string values.
+* **Default:** see description
+* **Example:** `ldap_search_fields = [{search_field = "User", ldap_field = "%u"}]`
 
-* `ldap_binary_search_fields` (list of binaries, default: `[]`): A list of search fields, which values should be Base64-encoded by MongooseIM before sending to LDAP.
+Mappings between the human-readable search fields and LDAP fields.
+For the default settings, please see `[MongooseIM root]/src/mod_vcard_ldap.erl`, line 101.
+
+###### `ldap_search_reported`
+* **Syntax:** Array of TOML tables with the following keys: `"search_field"`, `"vcard_field"` and string values.
+* **Default:** see description
+* **Example:** `ldap_search_reported = [{search_field = "Full Name", vcard_field = "FN"}]`
+
+Mappings between the human-readable search fields and VCard fields.
+For the default settings, please see `[MongooseIM root]/src/mod_vcard_ldap.erl`, line 114.
+
+###### `ldap_search_operator`
+* **Syntax:** string, one of `"or"`, `"and"`
+* **Default:** `"and"`
+* **Example:** `ldap_search_operator = "or"`
+
+A default operator used for search query items.
+
+###### `ldap_binary_search_fields`
+* **Syntax:** array of strings
+* **Default:** `[""]`
```

Comment:
Is this default value correct, considering that it is validated with `validate_non_empty_binary/1`?

Reply:
According to the code the default is `[]`, the validator is for items and verifies that each of them is a non-empty binary.

Reply:
I think the default kicks in when this is not set at all. In such case the default is an empty list.